diff --git a/src/gstreamer/gstlibcamera-utils.cpp b/src/gstreamer/gstlibcamera-utils.cpp
index 36b9564c..d6e9ac75 100644
--- a/src/gstreamer/gstlibcamera-utils.cpp
+++ b/src/gstreamer/gstlibcamera-utils.cpp
@@ -148,7 +148,7 @@ colorspace_from_colorimetry(const GstVideoColorimetry &colorimetry)
 	case GST_VIDEO_TRANSFER_GAMMA22:
 	case GST_VIDEO_TRANSFER_GAMMA28:
 		GST_WARNING("GAMMA 18, 20, 22, 28 transfer functions not supported");
-	/* fallthrough */
+		[[fallthrough]];
 	case GST_VIDEO_TRANSFER_GAMMA10:
 		colorspace->transferFunction = ColorSpace::TransferFunction::Linear;
 		break;
diff --git a/src/libcamera/camera_sensor.cpp b/src/libcamera/camera_sensor.cpp
index ae3127d6..a210aa4f 100644
--- a/src/libcamera/camera_sensor.cpp
+++ b/src/libcamera/camera_sensor.cpp
@@ -427,7 +427,7 @@ int CameraSensor::initProperties()
 			LOG(CameraSensor, Warning)
 				<< "Unsupported camera location "
 				<< v4l2Orientation << ", setting to External";
-			/* Fall-through */
+			[[fallthrough]];
 		case V4L2_CAMERA_ORIENTATION_EXTERNAL:
 			propertyValue = properties::CameraLocationExternal;
 			break;

